pandas/core/arrays/string_.py

```diff
@@ -753,8 +742,11 @@ def __setitem__(self, key, value) -> None:
                     "instead."
                 )
         else:
+            value = extract_array(value, extract_numpy=True)
             if not is_array_like(value):
                 value = np.asarray(value, dtype=object)
+            elif isinstance(value.dtype, type(self.dtype)):
```

Comment:
Should this use `is_string_dtype` to be permissive to other string types?

Reply:
Not at the moment, because if it passes here I only use this below for directly accessing the underlying array, and that only works if it is exactly the same dtype (in theory we could optimize for other string-like EAs to avoid validating that it are all strings, but I would consider that as a possible future optimization)